dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report.repo.test.ts

```diff
@@ -0,0 +1,52 @@
+import { when } from 'jest-when';
+import { Not } from 'typeorm';
+import { ReportPeriod, UtilisationReportEntity, UtilisationReportReconciliationStatus } from '@ukef/dtfs2-common';
+import { UtilisationReportRepo } from './utilisation-report.repo';
+
+describe('UtilisationReportRepo', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('findReportingYearsByBankId', () => {
+    const findSpy = jest.spyOn(UtilisationReportRepo, 'find');
+
+    const aUtilisationReportEntityWithOnlyReportPeriodSelected = (reportPeriod: ReportPeriod): UtilisationReportEntity =>
```

Comment:
Can we shorten some of the variable names, they are really long and hard to follow e.g

```ts
- aUtilisationReportEntityWithOnlyReportPeriodSelected
- aUtilisationReportForStatusBankIdAndMonthlyReportPeriod (what does `ForStatusBankId` mean?)
- getIncrementingPositiveIntegerIdGenerator
```
Maybe could be something more like this:

```ts
- utilisationReport({ selectedReportPeriods: 1 });
- utilisationReport({ monthlyReportPeriod: true });
- idGenerator
```

Reply:
Shorter variable names often lack any context and can therefore usually be extremely hard to follow. Especially considering that this is a test file, I don't see the downside to having these longer variable/function names

Reply:
I am finding the long variables names extremely hard to follow.

I worry that someone new (like me), could look at such functions/names in the future and find it hard to follow.

How else could this be improved without shortening the names?

Reply:
I don't believe the long variable names are hard to follow, and actually think that shorter variable names make things much harder. For example, as someone who joined this code base nearly a year ago, existing tests often were and still are very hard to follow due to the lack of descriptive variable names.

If the choice is between longer and shorter variable names with readability being cited as the threshold breaker, then I do believe longer variable names are the way to go

Reply:
What would the limit be for such long variable names, what happens if we want to add more params/conditions? It becomes unreadable e.g

```ts
aUtilisationReportEntityWithOnlyReportPeriodSelected
aUtilisationReportEntityWithOnlyReportPeriodSelectedWithXDate
aUtilisationReportEntityWithTwoReportPeriodsSelectedWithOnePastADeadline
aUtilisationReportEntityWithOneInvalidAndOneValidReportPeriodsSelected
```

Reply:
I don't think those names are necessarily unreadable, but then you can start citing context for a specific state if the variable name becomes unreasonably long. For an actual example, we have cases where we want to define a report to have fields which reflect that the report has been uploaded. Rather than calling the function `aUtilisationReportEntityWithUploadedByUserIdAndDateUploadedAnd...`, we call the function `anUploadedUtilisationReportEntity` which can accept custom parameters. It's all context specific, but in the context of this file the thing we care about is that the utilisation report only has a report period defined on it because we are selecting that field in the typeorm query, which is something that does not normally happen. The extra documentation we get for free that the entity yielded by this function is a report _only_ with a report period selected would be hard to document without context of how this function is implemented

Reply:
In specific case I would actually be tempted to just inline this as it is just one line and only used twice in quite a specific niche case and then avoids the need to try and come up with a good name? I think should be accompanied by a comment as well since it is a little odd and worth outlining it is intentional that we are using the `as`.